src/phoenix/db/bulk_inserter.py

```diff
@@ -75,6 +84,46 @@ async def _insert_spans(self) -> None:
             except Exception:
                 logger.exception("Failed to insert spans")
 
+    async def _insert_evaluations(self) -> None:
+        evaluations = self._evaluations
+        self._evaluations = []
+        for i in range(0, len(evaluations), self._max_num_per_transaction):
+            try:
+                async with self._db() as session:
+                    for evaluation in islice(evaluations, i, i + self._max_num_per_transaction):
+                        try:
+                            async with session.begin_nested():
+                                await _insert_evaluation(session, evaluation)
+                        except Exception:
+                            logger.exception(
+                                "Failed to insert evaluation "
+                                f"for span_id={evaluation.SubjectId.span_id}"
+                            )
+            except Exception:
+                logger.exception("Failed to insert evaluations")
+
+
+async def _insert_evaluation(session: AsyncSession, evaluation: pb.Evaluation) -> None:
+    if not (
+        span_rowid := await session.scalar(
+            select(models.Span.id).where(models.Span.span_id == evaluation.subject_id.span_id)
+        )
+    ):
+        return
```

Comment:
maybe this should be an exception
```suggestion
        raise ValueError(f"Span not found: {span_id=}")
```

Reply:
I don't think `session.begin_nested` is working as expected. When I raise an exception here, it kills the entire transaction and no subsequent evals are inserted. Going to leave as a `return` for now and file and issue to fix.